### DISCUSSION TITLE
Create Row Post API Not Functioning as Expected

### DISCUSSION THREAD

**Discussion by invictus2010** [nocodb/nocodb#1130] [General]
**Please enter the following details**

please provide the following details :
Node: **v12.22.6**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **true**
RootDB: **pg**
PackageVersion: **0.84.8**

**Steps To Reproduce**
1. Create new blank table called "Feedback"
2. Rename single column to "Feedback"
3. Add initial entry "HI there"
3. Generate API Token
4. Add API Token to POST request header as xc-token
5. Add string to body of request: "testing from Postman"
6. Send Post request
7. API response is "HI there." New row is not inserted 

**Expected behavior**
I would expect response to be 
[
{ 
"id": 2,
"Feedback": "testing from postman
...
}
]

**Screenshots**
![image](https://user-images.githubusercontent.com/17661001/152454530-7d9d4d53-b28e-4f70-81ee-ee7e6a11f5e2.png)

**Comment by wingkwong** [comment]
@invictus2010 I couldn't reproduce the issue. Could you share the version of your Postgres?

![image](https://user-images.githubusercontent.com/35857179/152482466-8975bd5e-5324-435f-ab7f-e3204bac9b6c.png)

**Comment by invictus2010** [comment]
I'm a little bit of a newbie. Specifically, I'm using Railway's insta-deploy of NoCoDB. I may try a self-install a little later today.

Meanwhile, I've gathered this when it comes to versions: `psql (14.1, server 13.2)`

**Comment by invictus2010** [comment]
Meanwhile, if its helpful this is what my Dockerfile looks like: 

```
FROM nocodb/nocodb:latest
ENTRYPOINT ["sh", "/usr/src/appEntry/start.sh"]
```

**Comment by invictus2010** [comment]
Here is what my logs look like when I send a POST request

```
select "nc_yyoj__feedback"."id" as "id", "nc_yyoj__feedback"."feedback" as "feedback", "nc_yyoj__feedback"."created_at" as "created_at", "nc_yyoj__feedback"."updated_at" as "updated_at" from "nc_yyoj__feedback"
select "nc_yyoj__feedback"."id" as "id", "nc_yyoj__feedback"."feedback" as "feedback", "nc_yyoj__feedback"."created_at" as "created_at", "nc_yyoj__feedback"."updated_at" as "updated_at" from "nc_yyoj__feedback"
GET /nc/haze_feedback_yyoj/api/v1/Feedback 200 2 - 7.438 ms
```

**Comment by invictus2010** [comment]
![image](https://user-images.githubusercontent.com/17661001/152556946-018959bf-7787-4ae4-92cf-9e9c1f42d569.png)
Tested it locally on SQLite and get "null" back. Inserts blank rows.

So maybe I'm just doing something wrong in my API calls?

**Comment by wingkwong** [comment]
@invictus2010 I'm using the same version. I'm wondering if you have included ``Content-Type: application/json`` in your header?

**Comment by invictus2010** [comment]
I hadn't! But that's still not fixing things...

Here's my request expressed in cURL

```curl 
curl --location --request POST 'http://haze-nocodb-production.up.railway.app/nc/haze_feedback_yyoj/api/v1/Feedback' \
--header 'xc-token: <token>' \
--header 'Content-Type: application/json' \
--data-raw '"feedback": "testing from postman" '
```

**Comment by wingkwong** [comment]
I believe the root cause for your original issue is missing ``Content-Type: application/json``.  For the request expressed in cURL, I notice that you were targeting to column ``feedback`` instead of ``Feedback``. 

Case 1: With ``Content-Type: application/json`` - Targeting to ``Feedback``.
```
curl -X 'POST' \ 
  'http://localhost:8080/nc/nocodb_7ltd/api/v1/Feedback' \
  -H 'accept: application/json' \
  -H 'xc-token: <REDACTED> ' \
  -H 'Content-Type: application/json' \
  -d '{
  "Feedback": "T2"
}'
```

Result: The data can be inserted as expected successfully.
```
{"Id":7,"Feedback":"T2","CreatedAt":"2022-02-08T11:36:47.065Z","UpdatedAt":"2022-02-08T11:36:47.065Z"}
```

![image](https://user-images.githubusercontent.com/35857179/152913734-ec3217b6-24ab-483d-b841-06aa26a39830.png)


Case 2: With ``Content-Type: application/json`` - Targeting to ``feedback`` instead of ``Feedback``.

```
curl -X 'POST' \ 
  'http://localhost:8080/nc/nocodb_7ltd/api/v1/Feedback' \
  -H 'accept: application/json' \
  -H 'xc-token: <REDACTED>' \
  -H 'Content-Type: application/json' \
  -d '{
  "feedback": "T2"
}'
```

Result: The data can be inserted but with NULL value.

```
{"Id":8,"Feedback":null,"CreatedAt":"2022-02-08T11:37:19.380Z","UpdatedAt":"2022-02-08T11:37:19.380Z"
```

![image](https://user-images.githubusercontent.com/35857179/152913897-c03989f4-102e-4b62-9227-bca62c5d0d69.png)

**Comment by o1lab** [comment]
Moving this to discussion
